### ISSUE TITLE
Surface graph problem

### ISSUE THREAD

**Issue by Pertrecho** [veusz/veusz#361]
Hello, I'm trying to create a surface graph from a set of data converted to XYZ from grid format (mesh). I've imported it as 1D sets and then created a 2D dataset, but the graphic created is not correct, and I don't know if there is information missing, or the conversion from grid to XYZ was not done correctly for Veusz to interpret the data correctly. I tried everything I could think of, but the closest I got to what I need is this: 

![RSM-Veusz](https://user-images.githubusercontent.com/41552146/71643186-055dbd00-2c95-11ea-92bf-6672b0f6825c.jpg)

instead of this (done in Excel):

![RSM-Excel](https://user-images.githubusercontent.com/41552146/71643189-0db5f800-2c95-11ea-81ff-ba7437f806f1.jpg)

Here I attach the data, in case you can help me solve this problem. Thanks.

[RSM data.zip](https://github.com/veusz/veusz/files/4014225/RSM.data.zip)

**Comment by jeremysanders** [comment]
I think you need to choose the "From x,y and z values based on 1D datasets or expressions" options. However, you don't have equal spacing of the coordinates on the axes, which this  currently doesn't support. I think this may be due to rounding of the coordinates.

**Comment by Pertrecho** [comment]
Thanks for your reply. I recalculated the spacing on the axes and got a surface graph, but it's a bit weird near the extremes, as it shows an upwards trend not present on the data. It looks to me as if Veusz is using 12 points of data, instead of 11 (looking at the number of lines per side), or like it's using each data point to draw the center of each square, instead of it's edges, and then giving a wrong angle to the squares on the edges of the surface (I hope my description makes sense):

![RSM-2](https://user-images.githubusercontent.com/41552146/71765216-6b825400-2ed0-11ea-8d8b-5d67044fb6af.jpg)

I attach the relevant files here:
[RSM-2.zip](https://github.com/veusz/veusz/files/4021867/RSM-2.zip)

**Comment by jeremysanders** [comment]
Try switching off the high resolution option on the 3d surface plotter. If that is set, then an extra interpolated node is placed at the centre of each part of the surface.

**Comment by Pertrecho** [comment]
Thanks again. High res. was off in the previous graph. With it on, it looks like this:
![RSM-3](https://user-images.githubusercontent.com/41552146/71768561-c9756280-2ef5-11ea-9b48-7d1177b41fd3.jpg)
It has an even mor noticeable "lip" around the edge.

**Comment by jeremysanders** [comment]
The coordinates define the centre of each of the shaded regions. The edges are calculated by interpolation.

**Comment by Pertrecho** [comment]
Exactly, that's what I thought. So, is there any way of getting rid of the "lip"? Because it misrepresents the trends of the rest of the graph.

**Comment by jeremysanders** [comment]
Only by extending your data. The current code only supports this method.

**Comment by Pertrecho** [comment]
Hi, thanks for your help. By extending the data to twice the points the graph improved a lot, as the outer squares are obviously smaller:

![RSM-3](https://user-images.githubusercontent.com/41552146/72207609-824d1b80-3479-11ea-9b3d-8f5147afd8cb.jpg)

Now that I know how to work around this current limitation, I'll leave it up to you to close this issue, or keep it open, for "inspiration", in case you are looking for features to improve.

Thanks a lot for your work with Veusz, is really a great program!
